frame/staking/src/pallet/impls.rs

```diff
@@ -863,6 +851,18 @@ impl<T: Config> Pallet<T> {
 			DispatchClass::Mandatory,
 		);
 	}
+
+	/// Perform all checks related to both [`Config::TargetList`] and [`Config::VoterList`].
+	#[cfg(debug_assertions)]
```

Comment:
Since this function is gated by `#[cfg(debug_assertions)]`, could we remove that same expression from the calls sites i.e. change

```rust
#[cfg(debug_assertions)]
Self::sanity_check_list_providers();
```

to simply

```rust
Self::sanity_check_list_providers();
```